File: app/src/main/java/com/maliks/applocker/xtreme/service/receiver/BootCompleteReceiver.java
package com.maliks.applocker.xtreme.service.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.util.Log;
import androidx.annotation.RequiresApi;
import com.maliks.applocker.xtreme.service.AppLockerService;

public class BootCompleteReceiver extends BroadcastReceiver {
    @RequiresApi(Build.VERSION_CODES.O)
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            Log.d("BootCompleteReceiver", "Boot completed. Starting AppLockerService.");

            Intent serviceIntent = new Intent(context, AppLockerService.class);
            context.startForegroundService(serviceIntent);
        }
    }
}

File: app/src/main/java/com/maliks/applocker/xtreme/service/receiver/BootCompleteReceiver.kt
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log
import androidx.annotation.RequiresApi
import com.maliks.applocker.xtreme.service.AppLockerService

class BootCompleteReceiver : BroadcastReceiver() {
    @RequiresApi(Build.VERSION_CODES.O)
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            Log.d("BootCompleteReceiver", "Boot completed. Starting AppLockerService.")
            
            val serviceIntent = Intent(context, AppLockerService::class.java)
            context.startForegroundService(serviceIntent)
        }
    }
}